src/overlays/actors/ovl_Bg_Jya_Ironobj/z_bg_jya_ironobj.c

```diff
@@ -43,21 +59,217 @@ static ColliderCylinderInit D_808994AC = {
     },
     { 30, 150, 0, { 0, 0, 0 } },
 };
+
+static s16 D_808994D8[] = { 0x8700, 0x4000, 0xC000, 0x0000 };
+
+// Pillar
+static s16 D_808994E0[] = { 0x0005, 0x0008, 0x000B, 0x000E, 0x0011, 0x0014, 0x0017, 0x001A };
+static s16 D_808994F0[] = { 0x0012, 0x001A, 0x0022, 0x002A, 0x0032, 0x003C, 0x0046, 0x0050 };
+static s16 D_80899500[] = { 0x0030, 0x002A, 0x0024, 0x0020, 0x001C, 0x0018, 0x0014, 0x0010 };
+
+// Throne
+static s16 D_80899510[] = { 0x0005, 0x0008, 0x000B, 0x000E, 0x0011, 0x0014, 0x0017, 0x001A };
+static s16 D_80899520[] = { 0x0012, 0x001A, 0x0022, 0x002A, 0x0032, 0x003C, 0x0046, 0x0050 };
+static s16 D_80899530[] = { 0x0030, 0x002A, 0x0024, 0x0020, 0x001C, 0x0018, 0x0014, 0x0010 };
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale.x, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+static CollisionHeader* sCollisionHeaders[] = { &gPillarCol, &gThroneCol };
+
+void BgJyaIronobj_InitCylinder(BgJyaIronobj* this, GlobalContext* globalCtx) {
+    ColliderCylinder* colCylinder;
+
+    colCylinder = &this->colCylinder;
+    Collider_InitCylinder(globalCtx, colCylinder);
+    Collider_SetCylinder(globalCtx, colCylinder, &this->dyna.actor, &sCylinderInit);
+    if ((this->dyna.actor.params & 1) == 1) {
+        this->colCylinder.dim.radius = 40;
+        this->colCylinder.dim.height = 100;
+    }
+    Collider_UpdateCylinder(&this->dyna.actor, colCylinder);
+}
+
+/*
+* Spawns particles for the destroyed pillar
+*/
+void BgJyaIronobj_SpawnPillarParticles(BgJyaIronobj* this, GlobalContext* globalCtx, EnIk* enIk) {
+    s32 i;
+    s32 j;
+    s16 unkArg5;
+    f32 temp_f22;
+    Vec3f pos;
+    Vec3f vel;
+    f32 coss;
+    s16 rotY;
+    f32 sins;
+    s32 pad[2];
+
+    if (enIk->unk_2FF <= 0 || enIk->unk_2FF >= 4) {
+        osSyncPrintf("Error 攻撃方法が分からない(%s %d)\n", "../z_bg_jya_ironobj.c", 233, enIk->unk_2FF);
+        return;
+    }
+    osSyncPrintf("¢ attack_type(%d)\n", enIk->unk_2FF);
+    rotY = Actor_WorldYawTowardActor(&this->dyna.actor, &enIk->actor) + D_808994D8[enIk->unk_2FF - 1];
+
+    for (i = 0; i < 8; i++) {
+        Actor* actor =
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_BG_JYA_HAHENIRON, this->dyna.actor.world.pos.x,
+                        Rand_ZeroOne() * 80.0f + this->dyna.actor.world.pos.y + 20.0f, this->dyna.actor.world.pos.z,
+                        0, ((s16)(s32)(Rand_ZeroOne() * 0x4000) + rotY) - 0x2000, 0, 0);
```

Comment:
```suggestion
                        0, ((s16)(Rand_ZeroOne() * 0x4000) + rotY) - 0x2000, 0, 0);
```
Double cast really necessary?